File: JornadaLaboralP/ClasesObjetos/JornadaLaboralAbs.py
class JornadaLaboral:
    """
    Esta clase se define como abstracta ya que solamente se establecerán los métodos para calcular
    el salario de un trabajador, cuyo método es el estático y en el cual se almacenarán los métodos
    pero con la diferencia de que no existirán parámetros debido a que en las clases abstractas no
    podemos crear instancias, así como se muestra en el método estático y solo retorna en el salario
    total.
    """
    #Método estático
    @staticmethod
    def calcularSalario (horasTrabajadas, horasExtras, salario):
        salarioTotal = horasTrabajadas * salario
        totalHorasExtras = horasExtras *(salario*1.5)
        salarioTotal += totalHorasExtras
        return  salarioTotal

File: JornadaLaboral/proyecto/JornadaLaboralAbs.java
package proyecto;

public class JornadaLaboralAbs {
/*
Para las clases abstractas, se definen ls métodos como se muestra a continuación, en este caso
se muestran dos métodos, en el primero se calcula el salario de un trabajador sin contar horas
extras, mientras que en el segundo si se utilizan. Al final se calcula el salario total y se
retorna el valor en ese atributo.
 */
    public static double calcularSalario(int horasTrabajadas, double salario) {
        double salarioTotal = horasTrabajadas * salario;
        return salarioTotal;
    }
    public static double calcularSalarioHrsExtras(int horasTrabajadas, double salario, int horasExtras){
        double salarioTotal = horasTrabajadas * salario;
        double totalHorasExtras = horasExtras *(salario*1.5);
        salarioTotal += totalHorasExtras;
        return salarioTotal;
    }

}

